gateway/sds_gateway/visualizations/cog_pipelines.py

```diff
@@ -0,0 +1,507 @@
+    """Get a pipeline configuration by type.
+
+    Args:
+        pipeline_type: Type of pipeline configuration to get
+
+    Returns:
+        Pipeline configuration dict
+    """
+    config_func = PIPELINE_CONFIGS.get(pipeline_type)
+    if not config_func:
+        error_msg = f"Unknown pipeline type: {pipeline_type}"
+        raise ValueError(error_msg)
+
+    return config_func()
+
+
+# Cog functions (pipeline steps)
+@cog
+def setup_post_processing_cog(capture_uuid: str, processing_types: list[str]) -> None:
+    """Setup post-processing for a capture.
+
+    Args:
+        capture_uuid: UUID of the capture to process
+        processing_types: List of processing types to run
+    Returns:
+        None
+    """
+
+    # imports to run when the app is ready
+    from sds_gateway.api_methods.models import Capture  # noqa: PLC0415
+    from sds_gateway.api_methods.models import CaptureType  # noqa: PLC0415
+
+    try:
+        logger.info(f"Starting setup for capture {capture_uuid}")
+
+        # Get the capture with retry mechanism for transaction timing issues
+        capture: Capture | None = None
+        max_retries = 3
+        retry_delay = 1  # seconds
+
+        for attempt in range(max_retries):
+            try:
+                capture = Capture.objects.get(uuid=capture_uuid, is_deleted=False)
+                break  # Found the capture, exit retry loop
+            except Capture.DoesNotExist:
+                if attempt < max_retries - 1:
+                    logger.warning(
+                        f"Capture {capture_uuid} not found on attempt {attempt + 1}, "
+                        f"retrying in {retry_delay} seconds..."
+                    )
+
+                    time.sleep(retry_delay)
+                    retry_delay *= 2  # Exponential backoff
+                else:
+                    # Final attempt failed
+                    error_msg = (
+                        f"Capture {capture_uuid} not found after {max_retries} attempts"
+                    )
+                    logger.error(error_msg)
+                    raise ValueError(error_msg) from None
+
+        # At this point, capture should not be None due to the retry logic above
+        assert capture is not None, (
+            f"Capture {capture_uuid} should have been found by now"
+        )
+
+        # Validate capture type
+        if capture.capture_type != CaptureType.DigitalRF:
+            error_msg = f"Capture {capture_uuid} is not a DigitalRF capture"
+            raise ValueError(error_msg)  # noqa: TRY301
+
+        # Set default processing types if not specified
+        if not processing_types:
+            error_msg = "No processing types specified"
+            raise ValueError(error_msg)  # noqa: TRY301
+
+        # Create PostProcessedData records for each processing type
+        for processing_type in processing_types:
+            _create_or_reset_processed_data(capture, processing_type)
```

Comment:
convert this `str` the enum before passing here, so it's easier to handle the types downwards e.g. `processing_type=ProcessingType(processing_type)`